data/model.py

```diff
@@ -0,0 +1,117 @@
+import abc
+
+import numpy
+from scipy.interpolate import interp1d
+
+import data.geo as geo
+
+
+class Model(metaclass=abc.ABCMeta):
+    """Abstract base class for models.
+    """
+
+    def __init__(self, nc_data):
+        self.interp: str = nc_data.interp
+        self.radius: int = nc_data.radius
+        self.neighbours: int = nc_data.neighbours
+
+    @property
+    @abc.abstractmethod
+    def depths(self) -> numpy.ndarray:
```

Comment:
Conceptually that makes sense, but because of `routes.api_v1_0.variables_query_v1_0()` calling `open_dataset(..., meta_only=True)` (above), if `depths` moves to `Data`, then we need an implementation in `NetCDFData`. I tried that, moving the implementation from `Mercator` to `NetCDFData`, and it fails with `AttributeError` because `self._dataset.coords` isn't defined yet in that `meta_only=True` case.

Reply:
Right and that goes back to me looking into `meta_only`. I'll tackle that today.

Reply:
Right so I don't really see an easy way to dump `meta_only` at the moment. My thinking is once this refactor is done, it may be easier to tease it out.